scripts/kernel_installer.py

```diff
@@ -0,0 +1,170 @@
+            r = requests.get(url, allow_redirects=True)
+            open('download.tar.gz', 'wb').write(r.content)
+            print('Done')
+
+            # Step 2: Untar
+            print('Step 2: Untar sources...')
+            my_tar = tarfile.open('download.tar.gz')
+            my_tar.extractall('.') # specify which folder to extract to
+            my_tar.close()
+            print('Done')
+
+            # Step 3: Apply EXTRAVERSION and patch
+            print('Step 3: Patch...')
+            with open('linux-5.10.35/Makefile') as f:
+                    data = f.read()
+                    data = data.replace('EXTRAVERSION =', f'EXTRAVERSION = .tfw-{current_patch_hash}')
+                    data = data.replace('NAME = Dare mighty things', f'NAME = Some test string')
+            with open(r'linux-5.10.35/Makefile', 'w') as file:
+                file.write(data)
+            shutil.copyfile(f'{tempesta_path}/linux-5.10.35.patch', 'linux-5.10.35/linux-5.10.35.patch')
+            with subprocess.Popen(["patch -p1 < linux-5.10.35.patch"],
+                                  stdout=subprocess.PIPE,
+                                  shell=True,
+                                  cwd=f"linux-5.10.35") as p:
+                output, err = p.communicate()
+                patch_out = output.decode("utf-8").strip()
+
+            # Step 4: config
+            if os.path.exists('.config'):
+                print('Step 4: Predefined .config found...')
+                subprocess.Popen(['cp', '.config', 'linux-5.10.35/config'], stdout=subprocess.PIPE)
+            else:
+                print('Step 4: Use OS .config...')
+                subprocess.Popen(['cp', f'/boot/config-{os.uname().release}', '.config'],
+                             stdout=subprocess.PIPE,
+                             cwd=f"linux-5.10.35")
+            
+            # Step 5: Build kernel
+            print('Step 5: Make deb-pkg...')
+            with subprocess.Popen(["make", "deb-pkg"],
+                                  stdout=subprocess.PIPE,
+                                  bufsize=1, universal_newlines=True,
+                                  cwd=f"linux-5.10.35") as p:
+                for line in p.stdout:
+                    print(line, end='') # process line here
+
+            if p.returncode != 0:
+                raise subprocess.CalledProcessError(p.returncode, p.args)
+
+            # Push to registry
+            if args.registry_creds is not None:
+                print('Step 5.1: Push to registry...')
+                for file in glob.glob('*.deb'):
+                    os.system(f'curl -u "{args.registry_creds}" \
+                            -H "Content-Type: multipart/form-data" --data-binary \
+                            "@./{file}" http://tempesta-vm.cloud.cherryservers.net:8081/repository/tempesta/')
+
+            # Step 6: Install and reboot
+            os.system('sudo dpkg -i *.deb')
+
+    if args.make_default:
```

Comment:
I'm not quite familiar with Debian-based distros, but on Fedora, one might just use `grubby --set-default=/boot/vmlinuz-5.10.35mykernel` to set the default kernel. Is there something similar for Ubuntu?

Reply:
For example `grubby` is not default package in Ubuntu